ee/clickhouse/queries/funnels/funnel_correlation.py

```diff
@@ -521,17 +538,101 @@ def _run(self) -> Tuple[List[EventOddsRatio], bool]:
+            # belongs in the application layer, or perhaps `FunnelCorrelationPeople`
+            params = self._filter.to_dict()
+            params = {
+                **params,
+                "events": json.dumps(params["events"]),
+                "funnel_correlation_person_converted": "true" if success else "false",
+                "funnel_correlation_person_entity": json.dumps(
+                    {"id": event_definition["event"], "type": "events", "properties": event_definition["properties"]}
+                ),
+            }
+            return f"/api/person/funnel/correlation/?{urllib.parse.urlencode(params)}"
+
+        if self._filter.correlation_type == FunnelCorrelationType.EVENT_WITH_PROPERTIES:
+            event_name, property_name, property_value = event_definition["event"].split("::")
+            params = self._filter.to_dict()
+            params = {
+                **params,
+                "events": json.dumps(params["events"]),
+                "funnel_correlation_event_names": json.dumps(params["funnel_correlation_event_names"]),
+                "funnel_correlation_person_converted": "true" if success else "false",
+                "funnel_correlation_person_entity": json.dumps(
+                    {
+                        "id": event_name,
+                        "type": "events",
+                        "properties": [
+                            {"key": property_name, "value": property_value, "type": "event", "operator": "exact"}
+                        ],
+                    }
+                ),
+            }
+            return f"/api/person/funnel/correlation/?{urllib.parse.urlencode(params)}"
+
+        if self._filter.correlation_type == FunnelCorrelationType.PROPERTIES:
+            # NOTE: for property correlations, we just use the regular funnel
+            # persons endpoint, with the breakdown value set, and we assume that
+            # event.event will be of the format "{property_name}::{property_value}"
+            # NOTE: this is just trying to recreate what we are doing in the
+            # frontend here:
+            # https://github.com/PostHog/posthog/blob/master/frontend/src/scenes/insights/InsightTabs/FunnelTab/FunnelPropertyCorrelationTable.tsx#L77:L77
+            #
+            # That's a pretty complicated code path so I'd have low confidence
+            # that this was right.
+            property_name, property_value = event_definition["event"].split("::")
+            params = {
+                key: value
+                for key, value in self._filter.to_dict().items()
+                # NOTE: we want to filter anything about correlation, as the
+                # funnel persons endpoint does not understand or need these
+                # params.
+                if not key.startswith("funnel_correlation_")
+            }
+            params = {
+                **params,
+                "events": json.dumps(params["events"]),
```

Comment:
I added a [utility function](https://github.com/PostHog/posthog/pull/6966/files#diff-7622fa5a28c0ece9e0a59cc1e684e72252cfed2df3d57fa531b72705d7dda805R132) in a different PR that should take care of these keys that already exist in params

Reply:
I've added a `to_params` as suggested elsewhere in the pr by @macobo